### PR TITLE
Create Shared content: Make impacts consistents across messenger secrets

### PR DESCRIPTION
**Pull request by loris-s-sonarsource**
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S6708/secrets/rule.adoc [hunk 1]
```diff
@@ -6,31 +6,17 @@ include::../../../shared_content/secrets/rationale.adoc[]
 
 === What is the potential impact?
 
-In the case of Discord, a webhook URL allows access to a channel and the
-consequences depend on what is specified in the `Bot Permissions`.
+The Discord webhook URL grants access to a channel to your code, represented by
```
Comment (daniel-teuchert-sonarsource) [inline rules/S6708/secrets/rule.adoc:9]:
```suggestion
The Discord webhook URL grants access to a channel in your server, represented by
```

File: shared_content/secrets/impact/personal_data_compromise.adoc [hunk 2]
```diff
@@ -0,0 +1,14 @@
+==== Compromise of sensitive personal data
+
+This kind of service is mainly used to exchange information that could include
```
Comment (daniel-teuchert-sonarsource) [inline shared_content/secrets/impact/personal_data_compromise.adoc:3]:
```suggestion
This kind of service is often used to exchange information that could include
```

File: shared_content/secrets/impact/phishing.adoc [hunk 3]
```diff
@@ -0,0 +1,6 @@
+==== Phishing and spam
+
+An attacker can use this {secret_type} to spam users or lure them into links to
```
Comment (daniel-teuchert-sonarsource) [inline shared_content/secrets/impact/phishing.adoc:3]: I did not know that we can use templates in our adocs. Where is {secret_type} refering to?

**Review by daniel-teuchert-sonarsource — APPROVED** [review]
LGTM! Minor comments and one question.

File: shared_content/secrets/impact/phishing.adoc (hunk 3 above)
Comment (loris-s-sonarsource) [reply shared_content/secrets/impact/phishing.adoc:3]: Basically here this is a [variable](https://docs.asciidoctor.org/pdf-converter/latest/theme/variables/).

This is what we filled here: https://github.com/SonarSource/rspec/blob/master/rules/S6699/secrets/rule.adoc?plain=1#L33-L35

Thanks for seeing it, I realize that I did not populate it in the files

File: rules/S6701/secrets/rule.adoc [hunk 4]
```diff
@@ -11,11 +11,14 @@ the Telegram Bot API. These keys are essentially access tokens that allow the
...
-primary concern is that unauthorized individuals may gain access to the bot's
-functionalities and data. This could result in misuse or abuse of the bot's
-capabilities. For instance, unauthorized users could send unsolicited
-messages, spam users, or engage in other disruptive activities using the bot.
+Below are some real-world scenarios that illustrate some impacts of an attacker
+exploiting the secret.
+
+include::../../../shared_content/secrets/impact/personal_data_compromise.adoc[]
+
+include::../../../shared_content/secrets/impact/phishing.adoc[]
```
Comment (loris-s-sonarsource) [inline rules/S6701/secrets/rule.adoc:19]:
```suggestion
:secret_type: secret


include::../../../shared_content/secrets/impact/phishing.adoc[]
```

File: rules/S6695/secrets/rule.adoc [hunk 5]
```diff
@@ -9,22 +9,16 @@ include::../../../shared_content/secrets/rationale.adoc[]
...
-actions. With the leaked app key, malicious actors could impersonate the app and
-perform actions on behalf of the app without proper authorization. This could
-lead to various security breaches, such as sending spam messages, spreading
-malware, or conducting phishing attacks on unsuspecting WeChat users.
+Below are some real-world scenarios that illustrate some impacts of an attacker
+exploiting the secret.
+
+include::../../../shared_content/secrets/impact/personal_data_compromise.adoc[]
+
+include::../../../shared_content/secrets/impact/phishing.adoc[]
```
Comment (loris-s-sonarsource) [inline rules/S6695/secrets/rule.adoc:17]:
```suggestion

:secret_type: secret

include::../../../shared_content/secrets/impact/phishing.adoc[]
```

File: rules/S6688/secrets/rule.adoc [hunk 6]
```diff
@@ -6,21 +6,22 @@ include::../../../shared_content/secrets/rationale.adoc[]
...
+* retrieving user data
+* posting on behalf of users
+* accessing various Facebook features
+
+Below are some real-world scenarios that illustrate some impacts of an attacker
+exploiting the secret.
+
+include::../../../shared_content/secrets/impact/personal_data_compromise.adoc[]
+
+include::../../../shared_content/secrets/impact/phishing.adoc[]
```
Comment (loris-s-sonarsource) [inline rules/S6688/secrets/rule.adoc:22]:
```suggestion
:secret_type: secret


include::../../../shared_content/secrets/impact/phishing.adoc[]
```

**Pull request merged**